posthog/api/test/dashboards/test_dashboard.py

```diff
@@ -147,6 +148,8 @@ def test_return_cached_results_bleh(self):
         self.assertAlmostEqual(Dashboard.objects.get().last_accessed_at, now(), delta=timezone.timedelta(seconds=5))
         self.assertEqual(response["tiles"][0]["insight"]["result"][0]["count"], 0)
 
+    # :KLUDGE: avoid making extra queries that are explicitly not cached in tests. Avoids false N+1-s.
```

Comment:
I'm concerned about this given I recently worked on it (though I need to resurrect the PR) - what is the 'false' N+1 being added here and is it truly false?

Reply:
Each query we make adds another insight that then gets `team.person_on_events_querying_enabled` called when it eventually gets queried as part of the dashboard.

I literally spent hours figuring out how to not get these extra queries for persons on events. This approach I settled on (exclude the N+1 tasks, add query counts elsewhere) is the least evil of all I tried. Snapshots won't work well either because each license test checks `valid_until > $inlined_timestamp`...